packages/assistify-dbs-translation/server/dbsTranslate.js

```diff
@@ -0,0 +1,219 @@
+					'name': TAPi18n.__('German', {lng: target})
+				},
+				{
+					'language': 'en',
+					'name': TAPi18n.__('English', {lng: target})
+				},
+				{
+					'language': 'fr',
+					'name': TAPi18n.__('French', {lng: target})
+				},
+				{
+					'language': 'es',
+					'name': TAPi18n.__('Spanish', {lng: target})
+				},
+				{
+					'language': 'it',
+					'name': TAPi18n.__('Italian', {lng: target})
+				},
+				{
+					'language': 'nl',
+					'name': TAPi18n.__('Dutch', {lng: target})
+				},
+				{
+					'language': 'pl',
+					'name': TAPi18n.__('Polish', {lng: target})
+				},
+				{
+					'language': 'ro',
+					'name': TAPi18n.__('Romanian', {lng: target})
+				},
+				{
+					'language': 'sk',
+					'name': TAPi18n.__('Slovak', {lng: target})
+				},
+				{
+					'language': 'ja',
+					'name': TAPi18n.__('Japanese', {lng: target})
+				},
+				{
+					'language': 'zh',
+					'name': TAPi18n.__('Chinese', {lng: target})
+				}
+			];
+		}
+	}
+
+	/**
+	 * Send Request REST API call to the service provider.
+	 * Returns translated message for each target language in target languages.
+	 * @private
+	 * @param {object} message
+	 * @param {object} targetLanguages
+	 * @returns {object} translations: Translated messages for each language
+	 */
+	_sendRequestTranslateMessage(message, targetLanguages) {
+		const translations = {};
+		const msgs = message.msg.split('\n');
+		const query = msgs.join();
+		let sourceLanguage;
+		//Since the translation provider do not handle the source language detection,
+		//we must handle explicitly say the source language
+		//We use language detection api to do it.
+		cld.detect(query, (err, result) => {
+			result.languages.map((language) => {
+				sourceLanguage = language.code || RocketChat.settings.get('Language') || 'en';
+			});
+		});
+		const supportedLanguages = this._getSupportedLanguages('en');
+		targetLanguages.forEach(language => {
+			if (language.indexOf('-') !== -1 && !_.findWhere(supportedLanguages, {language})) {
+				language = language.substr(0, 2);
+			}
+			try {
+				const result = HTTP.call('POST', `${ this.apiEndPointUrl }/translate`, {
+					params: {
+						key: this.apiKey
+					}, data: {
+						text: query,
+						to: language,
+						from: sourceLanguage
+					}
+				});
+
+				if (result.statusCode === 200 && result.data && result.data.translation && result.data.translation.length > 0) {
+					const txt = decodeURIComponent(result.data.translation);
+					translations[language] = this.deTokenize(Object.assign({}, message, {msg: txt}));
+				}
+			} catch (e) {
+				SystemLogger.error('Error translating message', e);
+			}
+		});
+		return translations;
+	}
+
+	/**
+	 * Returns translated message attachment description in target languages.
+	 * @private
+	 * @param {object} attachment
+	 * @param {object} targetLanguages
+	 * @returns {object} translated messages for each target language
+	 */
+	_sendRequestTranslateMessageAttachments(attachment, targetLanguages) {
```

Comment:
There's a lot of redundant code (language detection, result interpretation). Consider refactoring to common helper methods.